includes/multi-currency/Compatibility.php

```diff
@@ -87,12 +87,41 @@ public function override_selected_currency() {
 	}
 
 	/**
-	 * Checks to see if the widgets should be hidden.
+	 * Deprecated method, please use should_disable_currency_switching.
 	 *
 	 * @return bool False if it shouldn't be hidden, true if it should.
 	 */
 	public function should_hide_widgets(): bool {
-		return apply_filters( MultiCurrency::FILTER_PREFIX . 'should_hide_widgets', false );
+		wc_deprecated_function( __FUNCTION__, '6.5.0', 'Compatibility::should_disable_currency_switching' );
+		return $this->should_disable_currency_switching();
+	}
+
+	/**
+	 * Checks to see if currency switching should be disabled, such as the widgets and the automatic geolocation switching.
+	 *
+	 * @return bool False if no, true if yes.
+	 */
+	public function should_disable_currency_switching(): bool {
```

Comment:
Just out of curiousity, why did we need a second filter, and deprecate the first one?

Reply:
I replied about this in the other thread :D